### PR TITLE
feat: no access to location state in departures

### PR DESCRIPTION
**Pull request by adriawh**
~~#### This PR is dependent on https://github.com/AtB-AS/mittatb-app/pull/3982, as it handles the location alert~~

### 🆕 Feature:
Added an emptyState with a button on departures for when a user has not given location permission. When the button is pressed, it behaves as clicking the arrow in the search bar; it will prompt the user about location permissions. 

### 🤓 Code changes:
- Added button to emptyState
- Added conditional for emptyState or departures depending on location persmission

### 👀 Preview:

| iOS | Android |
|-----|---------|
|  |  |

### CODE REVIEW

File: src/nearby-stop-places/NearbyStopPlacesScreenComponent.tsx [hunk 1]
```diff
@@ -180,14 +185,37 @@ export const NearbyStopPlacesScreenComponent = ({
...
       <ScreenReaderAnnouncement message={loadAnnouncement} />
-      <StopPlaces
-        header={getListDescription()}
-        stopPlaces={orderedStopPlaces}
-        navigateToPlace={onSelectStopPlace}
-        testID="nearbyStopsContainerView"
-        location={location}
-        isLoading={isLoading}
-      />
+      {hasLocationPermission ? (
```
Comment (marius-at-atb) [inline src/nearby-stop-places/NearbyStopPlacesScreenComponent.tsx:188]:
```suggestion
      {hasLocationPermission || !!location ? (
```
Also check if a location has been selected from favorites

File: src/translations/screens/Nearby.ts [hunk 2]
```diff
@@ -23,6 +23,19 @@ const NearbyTexts = {
     },
   },
   stateAnnouncements: {
+    noAccessToLocationTitle: _(
+      'Holdeplasser i nærheten er ikke tilgjengelig.',
+      'Nearby departures are not available.',
+      'Holdeplassar i nærleiken er ikkje tilgjengelege.',
```
Comment (marius-at-atb) [inline src/translations/screens/Nearby.ts:29]:
```suggestion
      'Haldeplassar i nærleiken er ikkje tilgjengeleg.',
```

File: src/translations/screens/Nearby.ts [hunk 3]
```diff
@@ -23,6 +23,19 @@ const NearbyTexts = {
...
+    ),
+    noAccessToLocation: _(
+      'Del din posisjon for å finne holdeplasser i nærheten. Bruk søkefeltet for å finne andre holdeplasser.',
+      'Share your location to find nearby stops. Use the search field to find other stops.',
+      'Del posisjonen din for å finne haldeplassar i nærleiken. Bruk søkjefeltet for å finne andre haldeplassar.',
+    ),
+    sharePositionButton: {
+      title: _('Del din posisjon', 'Share your position', 'Del din posisjon'),
```
Comment (marius-at-atb) [inline src/translations/screens/Nearby.ts:37]:
```suggestion
      title: _('Del din posisjon', 'Share your position', 'Del posisjonen din'),
```

**Review by marius-at-atb — CHANGES_REQUESTED** [review]
Nice!
Just need to make sure that selecting a favorite location still works when no location permission granted.

File: src/nearby-stop-places/NearbyStopPlacesScreenComponent.tsx (hunk 1 above)
Comment (marius-at-atb) [reply src/nearby-stop-places/NearbyStopPlacesScreenComponent.tsx:188]: Flipping the condition could also be an option:
`{!hasLocationPermission ? <EmptyState ... /> : <StopPlaces ... />`

**Review by gorandalum — CHANGES_REQUESTED** [review]
Passing around React-components not as children are not really good practice, as it breaks the out-of-the-box shadow dom optimisations of React.

In addition the button inside the `EmptyState` component is probably ment to be styled in a certain way, and by opening up for passing in any button the component looses the ability to control its own design. I'm guessing we would never want a full width primary button in the `EmptyState` component.

The fix here is to change the input fields of the `EmptyState` component to be text strings and the onpress function, while creating the `Button` component and controlling the style of it is done inside the `EmptyState` component.

**Review by marius-at-atb — APPROVED** [review]
:+1:

**Review by gorandalum — APPROVED** [review]

**Pull request merged**